### ISSUE TITLE
Client Receive First Message

### ISSUE THREAD

**Issue by nahrens007** [redninja2/RSAServer#1]
When the client first connects to the server, the client does not receive the first message... The client does not receive a message until it sends something first. 

For the client, I have been using the client below. I have tested it with the java server as well, and it is capable of receiving messages before sending any. This must be a server side issue.
https://github.com/redninja2/ChatProgram/tree/master/src/plain/client

**Comment by nahrens007** [comment]
After further troubleshooting, the client will receive a message without sending anything first, just not right away. 
I connected 2 clients to server. 
1st client sent message and received its pub key as well as the message that it sent (which was broadcasted).
2nd client then received its pub key and the broadcasted message.

**Comment by nahrens007** [comment]
Nick, any idea why the client wouldn't be receiving the message until it sends something? I think it may have something to do with the server's send buffer?

**Comment by Shadow1356** [comment]
Off the top of my head and skimming the code, I can't think of anything. I'll do some testing tonight, and let you know what I find out.

**Comment by nahrens007** [comment]
Ok sweet. I tested this with my Java server (sending something to the client before the client sends anything) to just test the logic of the whole server stuff, but that was successful so I would think that the logic is good. I would think that the server would be able to send something to the client as soon as it gets past accept(), which is the case in the java server but not with python. I've noticed with my Java server that if I don't flush() the output stream, then the message isn't received by the client. But trying to flush() on the output stream in Python is a little different; I assume that has something to do with it. 
I'm going to create a simple python client and see if I get any results - so far I've only been using my Java client.
